packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -15,206 +15,53 @@ import {
-
-    /**
-     * Function call to handle keydown events,
-     */
-    onKeyDown: PropTypes.func,
-
-    /**
-     * Whether to keep the popout on screen when the anchor element scrolls off
-     */
-    keepOnScreen: PropTypes.bool,
-
-    /**
-     * Whether to lock the scrollbar when the popover is open
-     */
-    lockScroll: PropTypes.bool,
-
-    /**
-     * The marigins of the page the popover should respect
-     */
-    marginThreshold: PropTypes.number,
-
-    /**
-     * Function call to handle clickaway/ close events,
-     * if not provided the anchor element must be removed to clear the popover or use isOpen
-     */
-    onClose: PropTypes.func,
-
-    /**
-     * Styles to apply to the component
-     */
-    styles: PropTypes.object
-  };
-
-  static defaultProps = {
-    closeOnScrolled: true,
-    component: "span",
-    fullWidth: false,
-    isOpen: true,
-    lockScroll: true,
-    marginThreshold: 16,
-    styles: {}
-  };
-
-  constructor(props) {
-    super(props);
-    this.positionChangeEventHandler = debounce(this.updatePosition, 10);
-    this.handleCloseEvent = this.handleCloseEvent.bind(this);
-    this.state = {
-      scrollContainer: null,
-      scrollContainerStyle: null,
-      contentStyles: {
-        top: 0,
-        left: 0,
-        visibility: "hidden"
-      }
-    };
-
-    this.contentRef = React.createRef();
-  }
-
-  componentDidMount() {
-    const { anchorEl, lockScroll } = this.props;
-
-    window.addEventListener("resize", this.positionChangeEventHandler);
-    window.addEventListener("scroll", this.positionChangeEventHandler);
-    window.addEventListener("mousedown", this.handleCloseEvent);
-    window.addEventListener("keydown", this.handleCloseEvent);
-
-    if (anchorEl) {
-      this.manageScrollableParentEventListeners(false);
-    }
-
-    if (anchorEl && lockScroll) {
-      this.lockParentScroll();
-    }
-
-    this.setPositioningStyle();
-  }
-
-  componentDidUpdate(prevProps) {
-    const anchorChange =
-      prevProps.anchorEl === null && this.props.anchorEl !== null;
-
-    /* need to NOT check that previous prop was !isOpen
-     * in order to cause positioning of popover to be correct
-     * when it is opened by a controlled Tooltip being opened
-     */
-    const openChange = this.props.isOpen;
-    const anchorPositionChange = !(
-      prevProps.anchorPosition === this.props.anchorPosition
-    );
-    const contentPositionChange = !(
-      prevProps.contentPosition === this.props.contentPosition
-    );
-
-    if (anchorChange) {
-      // add event listeners
-      this.manageScrollableParentEventListeners(false);
-    }
-
-    if (
-      anchorChange ||
-      openChange ||
-      contentPositionChange ||
-      anchorPositionChange
-    ) {
-      // need to use requestAnimationFrame in order for popover to get proper position on first open
-      window.requestAnimationFrame(this.setPositioningStyle);
-    }
+const isDecendant = (parent, child) => {
```

Comment:
Little typo? `isDecendant` -> `isDescendant`

Reply:
Y'know, I just want to make sure my constants are up to Gordon Ramsay standards :P